src/pages/index.js

```diff
@@ -54,39 +72,136 @@ const profileEditModal = new ModalWithForm(
   handleProfileEditSubmit
 );
 profileEditModal.setEventListeners();
+
 const addCardModal = new ModalWithForm("#add-card-modal", handleAddCardCreate);
 addCardModal.setEventListeners();
 
+const profileAvatarEditModal = new ModalWithForm(
+  "#profile-avatar-edit-modal",
+  handleProfileAvatarUpdate
+);
+profileAvatarEditModal.setEventListeners();
+
+// Add an event listener to the button to open the avatar update modal
+const profileAvatarEditButton = document.querySelector(
+  "#profile-avatar-edit-button"
+);
+profileAvatarEditButton.addEventListener("click", () => {
+  profileAvatarEditModal.open();
+  profileAvatarInput.value = "";
+});
+
 /*    Form validators    */
-console.log(document.forms);
 const profileEditForm = document.forms["profile-edit-form"];
 const addCardForm = document.forms["add-card-form"];
+const profileAvatarForm = document.forms["profile-avatar-edit-form"];
 const editFormValidator = new FormValidator(config, profileEditForm);
 const addFormValidator = new FormValidator(config, addCardForm);
+const avatarFormValidator = new FormValidator(config, profileAvatarForm);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 
 /*    ModalWithImage Instances    */
 const previewImageModal = new ModalWithImage("#preview-image-modal");
 previewImageModal.setEventListeners();
 
+// Handle profile avatar update function
+function handleProfileAvatarUpdate(data) {
+  api
+    .updateUserAvatar(data.avatar)
+    .then((userData) => {
+      // Update the avatar image
+      const avatarElement = document.querySelector(".profile__avatar");
+      avatarElement.src = userData.avatar;
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+        avatar: userData.avatar,
+      });
+
+      profileAvatarEditModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+
+      const errorMessage = document.querySelector(
+        "#profile-avatar-input-error"
+      );
+      errorMessage.textContent = "Failed to update avatar. Please try again.";
+    })
+    .finally(() => {
+      profileAvatarEditModal.setModalLoad(false);
+    });
+}
+
 /*    Functions for opening/closing modals    */
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo({ name: data.name, description: data.description });
-  profileEditModal.close();
+  api
+    .updateUserInfo(data.name, data.description)
+    .then((userData) => {
+      // Update the UI with the new user data
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+        avatar: userData.avatar,
+      });
+
+      // Close the modal
+      profileEditModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      profileEditModal.setModalLoad(false);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```